src/components/themed/TransactionListTop.tsx

```diff
@@ -181,6 +206,180 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
+    bridge.resolve()
+  }
+
+  handleTradeSell = (bridge: AirshipBridge<void>) => {
+    const { navigation, wallet, tokenId } = this.props
+    const { pluginId } = wallet.currencyInfo
+
+    navigation.navigate('sellTab', { screen: 'pluginListSell', params: { filterAsset: { pluginId, tokenId } } })
+    bridge.resolve()
+  }
+
+  handleTradeSwap = (bridge: AirshipBridge<void>) => {
+    const { account, navigation, wallet, tokenId } = this.props
+
+    const sceneWallet = wallet
+    const sceneTokenId = tokenId
+    const { currencyWallets } = account
+    const { rootState } = this.props
+
+    // Check balances for the displayed asset on this scene:
+    const sceneAssetCryptoBalance = wallet.balanceMap.get(tokenId)
+
+    // Filter owned assets (mainnet and token) by those defined by the
+    // SWAP_ASSET_PRIORITY:
+    const ownedAssets: Array<{ pluginId: string; tokenId: EdgeTokenId; walletId: string; dollarValue: number }> = []
+    SWAP_ASSET_PRIORITY.forEach(priorityAsset => {
+      // Filter the currencyWallets for nonzero crypto balances that are
+      // NOT the asset shown on the scene
+      Object.values(currencyWallets).forEach(accountWallet => {
+        const isSceneAssetMatch =
+          accountWallet.currencyInfo.pluginId === sceneWallet.currencyInfo.pluginId &&
+          (sceneTokenId == null || accountWallet.enabledTokenIds.includes(sceneTokenId))
+
+        const isPriorityAssetMatch =
+          accountWallet.currencyInfo.pluginId === priorityAsset.pluginId && accountWallet.balanceMap.get(priorityAsset.tokenId) != null
+
+        if (!isSceneAssetMatch && isPriorityAssetMatch) {
+          const cryptoAmount = new CryptoAmount({
+            currencyConfig: accountWallet.currencyConfig,
+            tokenId: priorityAsset.tokenId,
+            nativeAmount: accountWallet.balanceMap.get(priorityAsset.tokenId) ?? '0'
+          })
+          const dollarValue = parseFloat(cryptoAmount.displayDollarValue(rootState, DECIMAL_PRECISION))
+
+          ownedAssets.push({
+            pluginId: priorityAsset.pluginId,
+            tokenId: priorityAsset.tokenId,
+            walletId: accountWallet.id,
+            dollarValue
+          })
+        }
+      })
+    })
+
+    let highestUsdBalanceAccountWalletId
+    let highestUsdBalancePriorityTokenId: EdgeTokenId = null
+
+    if (ownedAssets.length > 0) {
+      // Sort by highest dollar value
+      ownedAssets.sort((a, b) => b.dollarValue - a.dollarValue)
+
+      // Use the highest in the priority asset list as the other side of
+      // the swap. If we don't have exchange rates for one or more assets,
+      // we would still retain the ordering set forth by
+      // SWAP_ASSET_PRIORITY to use for this pick.
+      highestUsdBalanceAccountWalletId = ownedAssets[0].walletId
+      highestUsdBalancePriorityTokenId = ownedAssets[0].tokenId
+    } else {
+      // If no balances for the priority assets, we leave the opposing
```

Comment:
This comment doesn't match the spec. In the condition where the "other" wallet has zero balance but it is in the priority list, pick the asset highest on the priority list. Only in the case the user doesn't have ANY of the priority assets would they be required to pick one. A unit test would have made this easy to spot in development and in code review. I "think" code actually will pick an asset with zero balance, but the sort routine might not give you the first on the list since you don't know the sort algo when all entries are equal.

Reply:
As I mentioned in the comment above, the sort retains the ordering set forth by the original SWAP_ASSET_PRIORITY array. If all things are equal, the sort algo retains the original ordering.

```
const ownedAssets: Array<{ name: string; dollarValue: number }> = [
  { name: 'Asset A', dollarValue: 0 },
  { name: 'Asset B', dollarValue: 0 },
  { name: 'Asset C', dollarValue: 0 }
]
ownedAssets.sort((a, b) => b.dollarValue - a.dollarValue)
// [{"dollarValue": 0, "name": "Asset A"}, {"dollarValue": 0, "name": "Asset B"}, {"dollarValue": 0, "name": "Asset C"}]
```

Reply:
I revised the description in the task because QA also needs to be on the same page and I think there's still ambiguity in the task. 

Hopefully it's clear now to everyone.

Reply:
Updated conditions for `isSceneAssetMatch` and `isPriorityAssetMatch` so that
balanances are not considered when populating `ownedAssets`. Only owning the
asset and having the token enabled is required, and the `ownedAssets.sort()`
handles the requirement as described in the prior comment

Reply:
Where are the fix up commits? Please don’t say you merged them in.

Reply:
My last comment describes the change I made in the fixup to satisfy the requirement...

I just split up the last fixup into two to make it more obvious what the changes were.